### PR TITLE
Fix/dependencies

### LINKED ISSUE #1877: Update packages in hint and release a new version
# 🚀 Feature request

## Description

Right now `hint` depends on `configuration-web-recommended` `v5.0.0` but it should be `v6.0.0`.
We should update this reference (and any others) and do a new release.

### PR DESCRIPTION
**Pull request by molant**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [x] Added/Updated related documentation.
- ~~Added/Updated related tests.~~

## Short description of the change(s)

Create a script that updates the local dependencies and add it to the `release` script.
The release script should stop if there are uncommited changes so this should be used as a warning if something changes. Also nothing should change after finishing the release. Again if something does it should be considered as a warning and further investigation should be required.

Additionally this updates the local dependencies of some packages.

Partially fixes #1877

### CODE REVIEW

File: scripts/fix-dependencies.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+ * * `packages`: The content of each `package.json` indexed by `name`.
+ * * `paths`: The routes to each `package.json`. We can predict the
+ *   path because some packages don't follow the convention (e.g.:
+ *   `extension-browser`).
+ *
+ */
+const initialize = () => {
+    const packagesFiles = globby.sync([
+        '**/*/package.json',
+        '!**/*/fixtures'
```
Comment (antross) [inline scripts/fix-dependencies.js]: Should this just be a shallow match under `packages/*/package.json` to also avoid `node_modules` and other places we might have a `package.json` that doesn't relate to one of our packages?

File: scripts/fix-dependencies.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            /**
+             * `version` is "pinned" in the `package.json` so we need to add `^` to do the match.
+             *
+             * This means we could "downgrade" a `dependency` in a `package.json` if the workspace
+             * version is smaller. If this happens this means that something has gone terribly
+             * wrong before or the user should make sure to rebase with the latest `master`.
+             */
+            if (version !== `^${dependency.version}`) {
+                console.log(`updating ${dependencyName} from "${version}" to "^${dependency.version}" in ${pkg.name}.${property}`);
+                pkg[property][dependencyName] = dependency.version;
```
Comment (antross) [inline scripts/fix-dependencies.js]: Shouldn't updates include the leading carat `^`?
```suggestion
                pkg[property][dependencyName] = `^${dependency.version}`;
```

File: packages/configuration-development/package.json [hunk 3]
```diff
@@ -6,7 +6,7 @@
     "@hint/formatter-summary": "^2.0.1",
     "@hint/hint-axe": "^2.1.1",
     "@hint/hint-babel-config": "^2.1.3",
-    "@hint/hint-button-type": "^1.0.1",
+    "@hint/hint-button-type": "2.0.0",
```
Comment (antross) [inline packages/configuration-development/package.json]: This and others should have `^` (I put another comment where this should be fixed in the script).
```suggestion
    "@hint/hint-button-type": "^2.0.0",
```

File: scripts/fix-dependencies.js (hunk 1 above)
Comment (molant) [reply scripts/fix-dependencies.js]: That's why I have `gitignore: true` so it doesn't match anything under `node_modules`

File: scripts/fix-dependencies.js (hunk 2 above)
Comment (molant) [reply scripts/fix-dependencies.js]: Yes, I had it before but when I was refactoring I must have delete it 🤔 Good catch!

**Comment by molant** [comment]
@antross @sarvaje this should be ready for another review

**Pull request merged**